src/wallet/bitcoin.ts

```diff
@@ -15,152 +17,153 @@ export interface BitcoinWallet {
-    network: string
+    network: Network
   ): Promise<string>;
 
   broadcastTransaction(
     transactionHex: string,
-    network: string
+    network: Network
   ): Promise<string>;
 
   getFee(): string;
 }
 
 /**
- * Dummy Bitcoin wallet based on [bcoin]{@link https://github.com/bcoin-org/bcoin}.
+ * Instance of a bitcoind 0.19.1 wallet.
  *
  * This is to be used for demos, examples and dev environment only. No safeguards are applied, no data is written on
  * the disk.
  * This is not to be used for mainnet, instead, implement your own {@link BitcoinWallet}
  */
-export class InMemoryBitcoinWallet implements BitcoinWallet {
+export class BitcoindWallet implements BitcoinWallet {
   public static async newInstance(
-    network: string,
-    peerUri: string,
-    hdKey: string
-  ): Promise<InMemoryBitcoinWallet> {
-    const parsedNetwork = Network.get(network);
-
-    const logger = new Logger({
-      level: "warning"
-    });
-    const walletdb = new WalletDB({
-      memory: true,
-      witness: true,
-      logger,
-      network: parsedNetwork
-    });
-    const chain = new Chain({
-      spv: true,
-      logger,
-      network: parsedNetwork
-    });
-    const pool = new Pool({
-      chain,
-      logger
-    });
-
-    await logger.open();
-    await pool.open();
-    await walletdb.open();
-    await chain.open();
-    await pool.connect();
-
-    const wallet = await walletdb.create({
-      logger,
-      network: parsedNetwork,
-      master: hdKey
-    });
-
-    const account = await wallet.getAccount(0);
-
-    for (let i = 0; i < 100; i++) {
-      pool.watchAddress(await account.deriveReceive(i).getAddress());
-      pool.watchAddress(await account.deriveChange(i).getAddress());
+    httpUrl: string,
+    username: string,
+    password: string,
+    network: Network,
+    hdKey?: string,
+    rescan?: boolean
+  ): Promise<BitcoindWallet> {
+    const auth = { username, password };
+
+    const key = !!hdKey
+      ? bip32.fromBase58(hdKey)
+      : bip32.fromSeed(crypto.randomBytes(32), networks[network]);
+
+    // use HD key fingerprint as wallet name
+    const walletName = key.fingerprint.toString("hex");
+    const walletExists = await axios
+      .post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "listwallets"
+        },
+        { auth }
+      )
+      .then(res => res.data.result.includes(walletName));
+
+    if (!walletExists) {
+      await axios.post(
+        httpUrl,
+        {
+          jsonrpc: "1.0",
+          method: "createwallet",
+          params: [walletName]
+        },
+        { auth }
+      );
     }
 
-    pool.startSync();
-
-    pool.on("tx", (tx: any) => {
-      walletdb.addTX(tx);
-    });
-
-    pool.on("block", (block: any) => {
-      walletdb.addBlock(block);
-      if (block.txs.length > 0) {
-        block.txs.forEach((tx: any) => {
-          walletdb.addTX(tx);
-        });
+    let descriptor = `wpkh(${key.toBase58()}/0h/0h/*h)`;
```

Comment:
We just need to remember that we use the same derivation path in create-comit-app or make it optionally configurable. 

See here: 
https://github.com/comit-network/create-comit-app/blob/eac523791b23d4c5cbf3a4e6af1a5f9b8b901c25/scripts/src/docker/bitcoin.rs#L143-L148

I'm pro changing it in create-comit-app to `m/0'/0'/0'` as the otherone seemed to be really weired.

Reply:
The "really weird" one is BIP44 compliant though 😅

https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki

Although I didn't know about the cointype to be fair. We should probably use a different number there and not 0 or 1 because we are in regtest mode.